data-reporting-service/observation-service/src/main/java/gov/cdc/etldatapipeline/observation/service/ObservationService.java

```diff
@@ -0,0 +1,78 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class ObservationService {
+    private static final Logger logger = LoggerFactory.getLogger(ObservationService.class);
+
+    @Value("${spring.kafka.stream.input.observation.topic-name}")
+    private String observationTopic = "cdc.nbs_odse.dbo.Observation";
+
+    @Value("${spring.kafka.stream.output.observation.topic-name}")
+    public String observationTopicOutput = "cdc.nbs_odse.dbo.Observation.output";
+
+    @Value("${spring.kafka.stream.output.observation.topic-name-transformed}")
+    public String observationTopicOutputTransformed = "cdc.nbs_odse.dbo.Observation.output-transformed";
+
+    private final IObservationRepository iObservationRepository;
+    private String topicDebugLog = "Received Observation ID: {} from topic: {}";
+    private final KafkaTemplate<String, String> kafkaTemplate;
+    private final ProcessObservationDataUtil processObservationDataUtil;
+
+
+    @Autowired
+    public void processMessage(StreamsBuilder streamsBuilder) {
+        streamsBuilder.stream(observationTopic, Consumed.with(Serdes.String(), Serdes.String()))
+                .filter((k, v) -> v != null)
+                .mapValues((key, value) -> processObservation(value))
+                .filter((key, value) -> value != null)
+                .to(observationTopicOutput, Produced.with(Serdes.String(), Serdes.String()));
+    }
+
+    private String processObservation(String value) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            objectMapper.registerModule(new JavaTimeModule());
+            JsonNode jsonNode = objectMapper.readTree(value);
+            JsonNode payloadNode = jsonNode.get("payload");
+            if (payloadNode != null && payloadNode.has("after")) {
+                JsonNode afterNode = payloadNode.get("after");
+                if (afterNode != null && afterNode.has("observation_uid")) {
+                    String observationUid = afterNode.get("observation_uid").asText();
+                    logger.debug(topicDebugLog, observationUid, observationTopic);
+                    Optional<Observation> observationData = iObservationRepository.computeObservations(observationUid);
+                    if(observationData.isPresent()) {
+                        ObservationTransformed observationTransformed = processObservationDataUtil.transformObservationData(observationData.get());
+                        kafkaTemplate.send(observationTopicOutputTransformed, observationTransformed.toString());
+                    }
+                    return objectMapper.writeValueAsString(observationData.get());
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Error processing observation: {}", e.getMessage());
```

Comment:
@rshanmugam-enquizit is it possible to get the corresponding topic name and its Id during the error

Reply:
This one will be fixed with the other as a part of optimization